xapian-core/weight/tfidfweight.cc

```diff
@@ -89,12 +128,131 @@ TfIdfWeight::TfIdfWeight(const std::string &normals, double slope, double delta)
 	need_stat(DOC_LENGTH_MAX);
 	need_stat(UNIQUE_TERMS);
     }
+    switch (normalizations[0]) {
```

Comment:
You can pull this into a private initialiser method to avoid the duplication.

Reply:
I have used delegating constructors . This would help avoid other duplication (apart from switch) as well.